lib/src/matchers.dart

```diff
@@ -27,6 +27,7 @@ Matcher isRoute<T extends Object?>({
 }) {
   // Remove once `named` argument is removed.
   if (whereName == null && named != null) {
+    // Nullable parameter so assignment is OK
     // ignore: parameter_assignments
     whereName = equals(named);
   }
```

Comment:
I mean technically this is not OK because it is nullable, we are just ignoring it because whoever wrote it didn't want to make a local variable:

```suggestion
  Matcher? where = whereName;
  if (where == null && named != null) {
    where = equals(named);
  }
```

This could be a solution (and update all references to `whereName` in this block, or we should update the comment to reflect the real reason:

```suggestion
  if (whereName == null && named != null) {
    // Reassigning it to the parameter to make the code flow nicer.
    // ignore: parameter_assignments
    whereName = equals(named);
  }
```